src/Aggregation/DateHistogram.php

```diff
@@ -11,6 +11,59 @@ class DateHistogram extends Histogram
 {
     public const DEFAULT_TIMEZONE_VALUE = 'UTC';
 
+    private const CALENDAR_INTERVAL_VALUES = [
+        '1m',
+        'minute',
+        '1h',
+        'hour',
+        '1d',
+        'day',
+        '1w',
+        'week',
+        '1M',
+        'month',
+        '1q',
+        'quarter',
+        '1y',
+        'year',
+    ];
+
+    /**
+     * Set the interval by which documents will be bucketed.
+     *
+     * @param int|string $interval
+     *
+     * @return $this
+     */
+    public function setInterval($interval): Histogram
```

Comment:
This returns an instance of the current clas, not the parent one.
WDYT?

```suggestion
    public function setInterval($interval): DateHistogram
```

Reply:
I have to respect the parent signature, which is `self` and not `static`.

Reply:
I see, and we can not use that until php v8
What about altering the parent method (the docblock should be added to this method too, btw), to use
```/**
    * @return static
    */```

and removing the `: self` from the method signature?

Reply:
I removed `:self` to add the typehint in DateHistogram.

`@return $this` seems better than `@return static` I think.